ray-operator/Makefile

```diff
@@ -108,6 +108,8 @@ PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
 define go-get-tool
 @[ -f $(1) ] || { \
 set -e ;\
+GO_VERSION_1_17_OR_HIGHER=$(shell echo `go version | cut -d ' ' -f 3 | cut -c 3- | cut -d '.' -f1-2`\>=1.17 | bc) ;\
+if [[ $$GO_VERSION_1_17_OR_HIGHER -eq 1 ]]; then echo "The Go version you're using is 1.17 or higher. Using 'go install' instead of 'go get'"; GOBIN=$(PROJECT_DIR)/bin go install $(2); fi ;\
```

Comment:
This is very ugly. Please help me make it prettier.  I didn't know how to.

Also if this project doesn't support ≥ Go 1.18, we can close this for now and update the dev docs to say so?

Reply:
Please check https://github.com/ray-project/kuberay/pull/241 for the details why we didn't upgrade golang version. I think we can bump the dependency version together (k8s.io/xxx, golang, kubebuilder along with controller-runtime etc)

For shorterm, please help cut a new PR to update the doc on the limitation. Thanks!

Reply:
thanks! [PR here](https://github.com/ray-project/kuberay/pull/261)